lib/rspec/core/example.rb

```diff
@@ -143,8 +143,12 @@ def initialize(example_group_class, description, user_metadata, example_block=ni
 
         @example_group_instance = @exception = nil
         @clock = RSpec::Core::Time
+        @reporter = RSpec::Core::NullReporter.new
```

Comment:
This creates an extra object allocation per example.  Given that the null reporter is stateless, can we move the method missing def, etc to the class level so that we can use `RSpec::Core::NullReporter` itself as the null reporter?  Then we don't have to allocate an extra object for this.